include/blmc_robots/mathematics/polynome.hxx

```diff
@@ -0,0 +1,159 @@
+/**
+ * @file polynome_impl.hh
+ * @author Maximilien Naveau (maximilien.naveau@gmail.com)
+ * @brief Polynomes object for trajectories.
+ * @version 0.1
+ * @date 2019-11-06
+ * 
+ * @copyright Copyright (c) 2019
+ * 
+ */
+
+#pragma once
+
+#ifdef max
+#undef max
+#undef min
+#endif
+
+#include <iostream>
+#include <vector>
+#include "blmc_robots/mathematics/polynome.hpp"
+
+namespace blmc_robots
+{
+
+/**
+ * Polynome<ORDER> definitions
+ */
+
+template<int ORDER>
+Polynome<ORDER>::Polynome()
+{
+  coefficients_.fill(0.0);
+}
+
+template<int ORDER>
+Polynome<ORDER>::~Polynome()
+{
+}
+
+template<int ORDER>
+double Polynome<ORDER>::compute(double x)
+{
+  double res = 0.0;
+  double pt = 1.0;
+  for(int i = 0 ; i < coefficients_.size() ; ++i)
+  {
+      res += coefficients_[i] * pt;
+      pt *= x;
+  }
+  return res;
+}
+
+template<int ORDER>
+double Polynome<ORDER>::compute_derivative(double x)
+{
+  double res = 0.0;
+  double pt = 1.0;
+  for(int i = 1 ; i < coefficients_.size() ; ++i)
+  {
+      res += i * coefficients_[i] * pt;
+      pt *= x ;
+  }
+  return res;
+}
+
+template<int ORDER>
+double Polynome<ORDER>::compute_sec_derivative(double x)
+{
+  double res = 0.0;
+  double pt = 1.0;
+  for(int i = 2 ; i < coefficients_.size() ; ++i)
+    {
+      res += i * (i - 1) * coefficients_[i] * pt;
+      pt *= x;
+    }
+  return res;
+}
+
+template<int ORDER>
+void Polynome<ORDER>::get_coefficients(Coefficients &coefficients) const
```

Comment:
Would there be any drawback when returning the coefficients instead of using an output argument? It would make the method more intuitive to use.

Reply:
Yes you would have to create a stack memory, I would prefer people to manage the memory as cleanly as possible.